simulation/container.py

```diff
@@ -147,29 +157,50 @@ def simulation(self) -> None:
               and add new dead instance
         """
 
+        print(len(self.object_list))
         while self.is_alive():
+            start = time.time()
+            # Note: Move and can_infect parameter reset. Then sort.
             for instance in self.object_list:
                 x = random.uniform(-self.move_distance_length, self.move_distance_length)
                 y = random.uniform(-self.move_distance_length, self.move_distance_length)
                 instance.move(x, y, self.width, self.height)
+                instance.can_infect = True
+
+            self.object_list.sort(key=lambda parameter: parameter.x)
+            # --- end.
 
+            for counter, instance in enumerate(self.object_list, 1):
                 if instance.current_condition == "susceptible":
-                    for infected in self.object_list:
-                        if infected.current_condition == "infected":
-                            if instance.is_in_infection_area(infected.x, infected.y,
-                                                             infected.infection_range):
+                    for next_instance in itertools.islice(self.object_list, counter, len(self.object_list)):
+                        if next_instance.current_condition == "infected":
+                            if instance.is_in_infection_area(next_instance.x, next_instance.y,
+                                                             next_instance.infection_range):
                                 if instance.infect():
+                                    instance.can_infect = False
+                                    instance.current_condition = "infected"
+                                    break
+                            else:
+                                break
+
+                elif instance.current_condition == "infected" and instance.can_infect:
+                    for next_instance in itertools.islice(self.object_list, counter, len(self.object_list)):
```

Comment:
If I am not wrong, third argument by default is None and indicates the max length of passed iterator. We could avoid useless additional `len()` operation here.

Reply:
I checked this in documentation and islice() function has two variants one with two parameters (interable, stop), and second with (iterable, start, stop[, step]) so None is not default, by it can be used as max length as you write. I did not read this earlier, so now I will fix it